testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/userdata/ProfileSettingsActivity.kt

```diff
@@ -75,9 +83,20 @@ class ProfileSettingsActivity : BaseActivity() {
             val imageUri = data?.data
             setProfileImage(imageUri)
             profileUrl = imageUri.toString()
+            profileUrlBase64 = encodeImageForMiniApp(profileUrl)
         }
     }
 
+    private fun encodeImageForMiniApp(profileUrl: String): String {
+        val uri = Uri.parse(profileUrl)
+        val imageStream = contentResolver.openInputStream(uri)
+        val bitmap = BitmapFactory.decodeStream(imageStream)
+        val byteArrayOutputStream = ByteArrayOutputStream()
+        bitmap.compress(Bitmap.CompressFormat.PNG, 100, byteArrayOutputStream)
```

Comment:
Should we use `JPEG` here instead? Using PNG will give us a much larger file size. I'm worried this will affect the performance when Apps are retrieving the image, so it will give mini app developers an inaccurate view of this feature (most host apps will be using an HTTP URL here instead of base64 data).

Also, maybe we can set the quality to a lower value? Probably even `50` is okay. What do you think?

Reply:
I think it's reasonable, I have updated with `JPEG` with `50` quality.